### DISCUSSION TITLE
GraphQL parameters and response is not typed?

### DISCUSSION THREAD

**Discussion by some-user123** [Diizzayy/nuxt-graphql-client#241] [Q&A]
I'd followed the Quick start and used the SpaceX launches query used as an example.

1. Install `nuxt-graphql-client` (using `pnpm`)
2. Add it to modules and set Graphql host in `nuxt.config.ts`
3. Add `starlink.gql`:
```gql
query launches($limit: Int = 5) {
  launches(limit: $limit) {
    id
    launch_year
    mission_name
  }
}
```
4. Run `pnpm dev`
5. Use it in SFC:
```js
const { data } = await useAsyncGql({
  operation: 'launches',
  variables: { limit: 10 },
})
```

I stumbled upon two things:

- I get a hint in VSCode that the (only available) operation is `launches`. ✔ But I don't get any hint, which parameters would be available. Even passing `limit = 'foo'` does not raise a type error. Are parameters not typed❓
- `data` is of type `Ref<any>`. I would have expected something more elaborated based on the GraphQL definition. Is my expectation wrong or am I doing something wrong❓

**Comment by Diizzayy** [comment]
> But I don't get any hint, which parameters would be available

@some-user123  Certainly leads me to believe that the issue here is your IDE. I suggest ensuring that you have the [Volar extension](https://marketplace.visualstudio.com/items?itemName=Vue.volar) installed and configured correctly.

> Even passing `limit = 'foo'` does not raise a type error

This would only raise an error if `typescript.strict` is set to `true`.

> `data` is of type Ref<any>. I would have expected something more elaborated based on the GraphQL definition.

It should all in fact be typed properly. 

![Screenshot 2022-10-31 at 6 37 12 AM](https://user-images.githubusercontent.com/19627670/198990359-01a9e434-74ca-4493-a636-7b448bb9f12e.png)

![Screenshot 2022-10-31 at 6 37 56 AM](https://user-images.githubusercontent.com/19627670/198990377-7ce4ef49-b9f1-4b65-b92e-380a8b079969.png)

**Reply by some-user123** [reply]
That's great to hear. Then it is the solution I'm looking for. I just need to get it running... Thanks!

**Reply by Diizzayy** [reply]
@some-user123 Feel free to reach out to me on discord (Diizzayy#1964) if you need any additional help

**Reply by some-user123** [reply]
Got it working, thanks :-)

I like the module, but I'm missing two features:
- I don't get any support (hints etc.) when writing the GraphQL query.
- Other solutions (like https://github.com/babyfish-ct/graphql-ts-client) only run the generate step once and then dynamically type your query result. That's super nice.

Are there any plans to add that to nuxt-graphql-client or ideas how to achieve that with nuxt-graphql-client?

**Reply by Nicolas-Nmedia** [reply]
Hey guys ! Just ran into the same problem in VS Code where, sadly, I'm not getting any type hints when using gql functions, even though I have Volar installed.

> Got it working, thanks :-)

May I inquire how you managed to do so ? Did you configure volar any special way ?

Thanks for your help and a big thank you for this module !

**Comment by cdwmhcc** [comment]
@Diizzayy webstorm doesn't have any type hints.
